gsuite_driver/driver.py

```diff
@@ -269,18 +293,79 @@ def all(self):
         self.drive_list = drives
         return drives
 
+    def archive(self, grouplist):
+        """Enumerate all teamDrive objects."""
+        interactive_result = ask_question(
+            interactive_mode=self.interactive_mode,
+            message='Run drive achiving routine?',
+            operation='FUNC',
+            drive_name=self.drive_name,
+            detail=None
+        )
+
+        if interactive_result is False:
+            return None
+
+        if self.gsuite_api is None:
+            self.authenticate()
+
+        all_drives = self.all()
+        archive = Archive(grouplist, all_drives, self.interactive_mode)
+        for drive in all_drives:
+            if archive.should_be_archived(drive):
+                self.drive = drive
+                self.drive_name = drive.get('name')
+                interactive_result = ask_question(
+                    interactive_mode=self.interactive_mode,
+                    message='Archive this drive?',
+                    operation='GET',
+                    drive_name=self.drive_name,
+                    detail=None
+                )
+
+                memberships = self._membership_to_email_list(self.members)
+
+                for member in memberships:
+                    interactive_result = ask_question(
+                        interactive_mode=self.interactive_mode,
+                        message='Remove user for drive?',
+                        operation='DEL',
+                        drive_name=self.drive_name,
+                        detail=member
+                    )
+                    self.member_remove(member)
+
+                if len(self.members) == 0:
+                    new_name = 'archived_' + str(int(time.time())) + '_' + drive.get('name')
```

Comment:
is a dot ok in the name? `archived_31337.31337_drivename` if yes no problem

Reply:
dots are fine.